### PR TITLE
lightgbm: add version 3.3.2

### PR DESCRIPTION
**Pull request by buresu**
Specify library name and version:  **lightgbm/3.3.2**

LightGBM is a machine learning framework developed by Microsoft.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by buresu** [comment]
Local tests pass with the fallowing commands, but not on CI.
```
export CC=/usr/bin/clang && export CXX=/usr/bin/clang++ && export CMAKE_C_COMPILER=/usr/bin/clang && export CMAKE_CXX_COMPILER=/usr/bin/clang++ && conan create conanfile.py lightgbm/3.3.2@ -k -s compiler=clang -s compiler.libcxx=libstdc++ -s compiler.version=12
```

File: recipes/lightgbm/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
```
Comment (uilianries) [inline recipes/lightgbm/all/conanfile.py:89]:
```suggestion
    def package_info(self):
        self.cpp_info.set_property("cmake_file_name", "LightGBM")
        self.cpp_info.set_property("cmake_target_name", "LightGBM::LightGBM")
        # TODO: to remove in conan v2 once cmake_find_package* generators removed
```

**Review by uilianries — COMMENTED** [review]
Very good recipe, only small points

File: recipes/lightgbm/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+required_conan_version = ">=1.33.0"
+
+
+class LightGBMConan(ConanFile):
+    name = "lightgbm"
+    description = "A fast, distributed, high performance gradient boosting (GBT, GBDT, GBRT, GBM or MART) framework based on decision tree algorithms, used for ranking, classification and many other machine learning tasks."
+    topics = ("machine-learning", "boosting")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/LightGBM"
+    license = "MIT License"
```
[the remark this answers is not in the document]
Comment (buresu) [reply recipes/lightgbm/all/conanfile.py]: I fixed it. Thank you for the information.

File: recipes/lightgbm/all/conanfile.py (hunk 1 above)
Comment (buresu) [reply recipes/lightgbm/all/conanfile.py:89]: I fixed it.

**Review by SSE4 — APPROVED** [review]

**Pull request merged**